File: Student_attendance/src/main/java/com/example/Student_attendance/Student_model/Student.java
package com.example.Student_attendance.Student_model;

import com.fasterxml.jackson.annotation.JsonSubTypes;
import jakarta.annotation.Nullable;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "students")
@AllArgsConstructor
@NoArgsConstructor
public class Student {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(nullable = false)
    private String name;
    @Getter
    @Column(nullable = false,unique = true)
    private String email;
    private String phoneNumber;
    @Column(nullable = false)
    private String password;
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Role role;

//    public static JsonSubTypes.Type builder() {
//    }

//    private String photoPath;
public enum Role {
    STUDENT,
    ADMIN
}
//
//    public Object getPassword() {
//        return password;
//    }
}

File: backend/models/Student.js
const mongoose = require("mongoose");

const studentSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
  },

  email: {
    type: String,
    required: true,
    unique: true,
  },

  password: {
    type: String,
    required: true,
  },

  college: {
    type: String,
    required: true,
  },

  branch: {
    type: String,
    required: true,
  },

  photo: {
    type: String,
  },

  role: {
    type: String,
    default: "student",
  },
});

module.exports = mongoose.model("Student", studentSchema);